drivers/input/touchscreen/rpi-ft5406.c

```diff
@@ -61,13 +60,14 @@ static int ft5406_thread(void *arg)
 	struct ft5406 *ts = (struct ft5406 *) arg;
 	struct ft5406_regs regs;
 	int known_ids = 0;
-	
+        uint8_t temp = 99;	
+
 	while(!kthread_should_stop())
 	{
 		// 60fps polling
 		msleep_interruptible(17);
-		memcpy_fromio(&regs, ts->regs, sizeof(*ts->regs));
-		writel(99, &ts->regs->num_points);
+		memcpy_fromio(&regs, ts->ts_base, sizeof(regs));
```

Comment:
I don't totally understand it myself.  I just know that the writel breaks on 64 bit.  I think it has something to do with the value only being 8 bit.